runners/flink/src/main/java/org/apache/beam/runners/flink/translation/wrappers/SourceInputFormat.java

```diff
@@ -108,12 +110,27 @@ public float getAverageRecordWidth() {
     return null;
   }
 
+  private long getDesiredSizeBytes(int numSplits) throws Exception {
+    long totalSize = initialSource.getEstimatedSizeBytes(options);
+    long defaultSplitSize = totalSize / numSplits;
+    if (initialSource instanceof FileBasedSource) {
+      long maxSplitSize = options.as(FlinkPipelineOptions.class).getFileInputSplitMaxSizeBytes();
```

Comment:
Could we let a value of 0 or -1 disable this new behavior and restore to the original?

Reply:
Sounds reasonable. I'll make the change.